README.md

```diff
@@ -4,4 +4,63 @@
 
 # Matryoshka
 
-This library provides generalized folds, unfolds, and traversals for fixpoint data structures in Scala.
+This library provides generalized folds, unfolds, and traversals for fixed point data structures in Scala.
+
+## External Resources
+
+- [Functional Programming with Bananas, Lenses, Envelopes and Barbed Wire](http://citeseerx.ist.psu.edu/viewdoc/summary?doi=10.1.1.41.125) – the iconic paper that collected a lot of this info for the first time
+- [Recursion Schemes: A Field Guide (Redux)](http://comonad.com/reader/2009/recursion-schemes/) – Ed Kmett’s summary (with links to Haskell code) of various folds and unfolds
+- [Unifying Structured Recursion Schemes](http://www.cs.ox.ac.uk/people/jeremy.gibbons/publications/urs.pdf) – a newer paper on how to generalize recursion schemes
+- [Efficient Nanopass Compilers using Cats and Matryoshka](https://github.com/sellout/recursion-scheme-talk/blob/master/nanopass-compiler-talk.org) – Greg Pfeil’s (Scala) talk on this library (and some other things)
+- [Fix Haskell (by eliminating recursion)](https://github.com/sellout/recursion-scheme-talk/blob/master/recursion-scheme-talk.org) – Greg Pfeil’s (Haskell) talk on recursion schemes
+
+## Introduction
+
+This library is predicated on the idea of rewriting your recursive data structures as functors where the recursive component is replaced by a parameter.
+
+```scala
+sealed trait Expr
+final case class Var(name: String)          extends Expr
+final case class Lam(v: String, body, Expr) extends Expr
+final case class App(f: Expr, arg: Expr)    extends Expr
+```
+could be rewritten as
+
+```scala
+sealed trait Expr[A]
+final case class Var[A](name: String)       extends Expr[A]
+final case class Lam[A](v: String, body, A) extends Expr[A]
+final case class App[A](f: A, arg: A)       extends Expr[A]
+```
+
+Then you use one of the fixed point type constructors below to regain your recursive type.
+
+### Fixpoint Types
+
+These types take a one-arg type constructor (a functor) and provide a recursive form of it.
```

Comment:
For the uninitiated, I think a not-quite trivial example with `Fix` would be a big help here, because it's the one you need to understand first to convert an existing type.

How about defining `ExprF` above and then here showing `type Expr = Fix[ExprF]`, and maybe constructing a value?